crates/subspace-node/src/bin/subspace-node.rs

```diff
@@ -482,7 +482,9 @@ fn main() -> Result<(), Error> {
                         maybe_secondary_chain_spec.ok_or_else(|| {
                             "Primary chain spec must contain secondary chain spec".to_string()
                         })?,
-                        cli.secondary_chain_args.iter(),
+                        [SecondaryChainCli::executable_name()]
+                            .iter()
+                            .chain(cli.secondary_chain_args.iter()),
```

Comment:
This is a nice usability improvement (though it'll may break our deployments where we use `-- --` already). An alternative would be to have this hack in `SecondaryChainCli` itself, so that public API is the same: just argument without executable name.

Reply:
Our deployment is already broken for secondary chain since we seem to be using only one `--` but luckily the chainspec args here https://github.com/subspace/infra/blob/main/network-primitives/scripts/create_rpc_node_compose_file.sh#L93 is useless as this is for sure ignored . While We use two `--` for core domains. If you mean older deployments, not sure if they will be long run. As for Gemini-3a, we are at the stage where executors are not deployed yet so this should not break it

Sure. we can move this inside the SecondarChainCli